modules/core/src/v2/coins/abstractEthLikeCoin.ts

```diff
@@ -0,0 +1,157 @@
+/**
+ * @prettier
+ */
+import * as Bluebird from 'bluebird';
+import { CoinFamily, BaseCoin as StaticsBaseCoin } from '@bitgo/statics';
+
+import {
+  BaseCoin,
+  KeyPair,
+  ParsedTransaction,
+  ParseTransactionOptions,
+  SignedTransaction,
+  SignTransactionOptions,
+  VerifyAddressOptions,
+  VerifyTransactionOptions,
+  TransactionFee,
+  TransactionRecipient as Recipient,
+  TransactionPrebuild as BaseTransactionPrebuild,
+  TransactionExplanation,
+} from '../baseCoin';
+
+import { BitGo } from '../../bitgo';
+import { NodeCallback } from '../types';
+import { MethodNotImplementedError } from '../../errors';
+
+export interface EthSignTransactionOptions extends SignTransactionOptions {
+  txPrebuild: TransactionPrebuild;
+  prv: string;
+}
+
+export interface TxInfo {
+  recipients: Recipient[];
+  from: string;
+  txid: string;
+}
+
+export interface TransactionPrebuild extends BaseTransactionPrebuild {
+  txHex: string;
+  txInfo: TxInfo;
+  feeInfo: EthTransactionFee;
+  source: string;
+  dataToSign: string;
+}
+
+export interface EthTransactionFee {
+  fee: string;
+  gasLimit?: string;
+}
+
+export interface ExplainTransactionOptions {
+  txHex?: string;
+  halfSigned?: {
+    txHex: string;
+  };
+  feeInfo: TransactionFee;
+}
+
+export class AbstractEthLikeCoin extends BaseCoin {
```

Comment:
We already have `AbstractUtxoCoin`, seems like the equivalent would be `AbstractAccountCoin`. Any reasons why you picked `EthLike` for the name?

Reply:
Because CELO, RSK and ETC are all either code clones or API-compatible with ETH, making the code for those 3 specifically reusable compared to other account-based coins which have different RPC calls, wallet formats etc. It's not super clear since there's no real implementation here, but once it is -- you'll see that it is logic that would work for ETH-like coins but not account-based coins in general